feature-tests/src/test/kotlin/fi/vauhtijuoksu/vauhtijuoksuapi/featuretest/GameDataTest.kt

```diff
@@ -0,0 +1,74 @@
+package fi.vauhtijuoksu.vauhtijuoksuapi.featuretest
+
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.authentication.UsernamePasswordCredentials
+import io.vertx.ext.web.client.WebClient
+import io.vertx.junit5.VertxExtension
+import io.vertx.junit5.VertxTestContext
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(VertxExtension::class)
+class GameDataTest {
+    lateinit var client: WebClient
+
+    private val gameData1 = """
+        {            
+            "game": "Tetris2",
+            "player": "jsloth",
+            "start_time": "2021-09-21T15:05:47.000+00:00",
+            "end_time": "2021-09-21T16:05:47.000+00:00",
+            "category": "any%",
+            "device": "PC",
+            "published": "1970",
+            "vod_link": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
+            "img_filename": "tetris.png",
+            "player_twitch": "jiisloth"
+        }
+    """.trimIndent()
+
+    @BeforeEach
+    fun setup() {
+        client = WebClient.create(Vertx.vertx())
+    }
+
+    @Test
+    fun `test post and patch`(testContext: VertxTestContext) {
+        client.post("/gamedata")
+            .authentication(UsernamePasswordCredentials("vauhtijuoksu", "vauhtijuoksu"))
+            .sendJson(JsonObject(gameData1))
+            .onFailure(testContext::failNow)
+            .onSuccess { res ->
+                testContext.verify {
+                    assertEquals(201, res.statusCode())
+                    val resJson = res.bodyAsJsonObject()
+                    val original = JsonObject(gameData1)
+                    original.put("id", resJson.getString("id"))
+                    assertEquals(original, res.bodyAsJsonObject())
+                    val id = resJson.getString("id")
+                    client.patch("/gamedata")
```

Comment:
I think it would be cleaner to add compose here and test the patch in a separate handler, to reduce indentation, and more clearly separate the second api call. Something like this (though not tested)
```suggestion
}
.compose {
                    client.patch("/gamedata")
```

Reply:
I don't think you can compose after success